### PR TITLE
PLATFORM-3537: upgrade Vignette images embedded in wikitext to use HTTPS

### PR DESCRIPTION
**Pull request by harnash**
When dealing with Vignette (-like) url embedded in wikitext we should upgrade them to HTTPS to avoid mixed content issues.

ping: @Wikia/core-platform-team

### CODE REVIEW

File: extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php [hunk 1]
```diff
@@ -68,4 +70,14 @@ private static function httpsEnabledTitle( Title $title ): bool {
 		return array_key_exists( $wgDBname, self::$httpsArticles ) &&
 			in_array( $title->getPrefixedDBKey(), self::$httpsArticles[ $wgDBname ] );
 	}
+
+	public static function parserUpgradeVignetteUrls ( string &$url ): bool {
+		if ( preg_match( self::VIGNETTE_IMAGES_HTTP_UPGRADABLE, $url ) && strpos( $url, "http://" ) === 0 ) {
+			$url = wfHttpToHttps( $url );
+			return true;
```
Comment (mszabo-wikia) [inline extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php]: Other way around—`false` aborts further hook processing, `true` doesn't 😉

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php (hunk 1 above)
Comment (harnash) [reply extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php]: oh... right! Fix: 11bef317cc

**Review by mszabo-wikia — APPROVED** [review]

File: extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php [hunk 2]
```diff
@@ -68,4 +70,10 @@ private static function httpsEnabledTitle( Title $title ): bool {
 		return array_key_exists( $wgDBname, self::$httpsArticles ) &&
 			in_array( $title->getPrefixedDBKey(), self::$httpsArticles[ $wgDBname ] );
 	}
+
+	public static function parserUpgradeVignetteUrls ( string &$url ): bool {
```
Comment (Grunny) [inline extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php]: This is no longer returning a bool, so the type declaration on the return value should be removed.

**Review by Grunny — CHANGES_REQUESTED** [review]

File: extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php (hunk 2 above)
Comment (harnash) [reply extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php]: fix: 8e855226164

**Review by Grunny — APPROVED** [review]

**Pull request merged**